lib/palindrome_check.rb

```diff
@@ -1,7 +1,38 @@
-# A method to check if the input string is a palindrome.
-# Return true if the string is a palindrome. Return false otherwise.
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n), where n represents the length of
+# my_phrase.
+# Space complexity: O(1) because you're just comparing the
+# string characters to eachother, then returning either
+# true or false.
+
 def palindrome_check(my_phrase)
-  raise NotImplementedError
+  if my_phrase == nil
+    return false
+  end
+  len = my_phrase.length
+  i = 0
+  j = 0
+  low = nil
+  high = nil
+  (len / 2).times do
+    if my_phrase[i] != " "
+      low = my_phrase[i]
+    else
+      i += 1
+    end
+    if my_phrase[len - 1 - j] != " "
```

Comment:
This math is a little more complicated than you need, maybe instead of i & j, make variables named `low_index` and `high_index` and start them at 0 and length -1.  It would be easier to read and maintain.

You could also make these if statements loops and that would help dry up your code.